vite-plugin-cjs-interop/src/index.ts

```diff
@@ -61,53 +70,119 @@ export function cjsInterop(options: CjsInteropOptions): Plugin {
 			});
 
 			const toBeFixed: any[] = [];
-			const dynamicImportsToBeFixed: any[] = [];
 			const preambles: string[] = [];
+			let hasDynamicImportsToFix = false;
 
 			const { walk } = await walker;
 
 			walk(ast as any, {
 				enter(node) {
-					if (node.type === "ImportDeclaration") {
-						if (matchesDependencies(node.source.value as string)) {
-							toBeFixed.push(node);
-						}
-					} else if (node.type === "ImportExpression") {
-						if (
-							node.source.type === "Literal" &&
-							matchesDependencies(node.source.value as string)
-						) {
-							dynamicImportsToBeFixed.push(node);
-						}
+					switch (node.type) {
+						case "ImportDeclaration":
+						case "ExportNamedDeclaration":
+						case "ImportExpression":
+							if (
+								node.source &&
+								node.source.type === "Literal" &&
+								matchesDependencies(node.source.value as string)
+							) {
+								toBeFixed.push(node);
+								if (node.type === "ImportExpression") {
+									hasDynamicImportsToFix = true;
+								}
+							}
+							break;
+						default:
```

Comment:
This adds support for `ExportNamedDeclaration` and applies the same logic to all nodes to be fixed to avoid duplicating code.